raiden-network/light-client
# Fix CLI bundling and export artifact in CI

**Thank you for submitting this pull request :)**

Fixes #2172 

**Short description**
Bundling was broken since `yargs@16` upgrade, which isn't working well right now, so we downgraded back to `yargs@15.4.1`. This includes some fixes to allow bundling to work with the new `leveldown` default database adapters (since #2110) by replacing `webpack` with [ncc](https://github.com/vercel/ncc) for bundling , tests the bundling on CI and export the resulting artifact.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. `pnpm run build:bundle` on `raiden-cli`
2. Check the bundle works fine even outside its folder (no external dependency besides nodeJS)
